src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -184,6 +184,18 @@ internal GrpcChannel(Uri address, GrpcChannelOptions channelOptions) : base(addr
         {
             Log.AddressPathUnused(Logger, Address.OriginalString);
         }
+
+        // Validate the Windows version can support WinHttpHandler.
+        const int WinServer2022BuildVersion = 20348;
```

Comment:
I think the comment needs to elaborate on why this particular build version was chosen.
Basically, as you said
- If OS build version is greater or equal to WinServer 2022 then there is partial support.
- If OS build version is greater or equal to Win 11 then there is full support.

The also need to be a note that this will allow creating GrpcChannel on WinServer 2022 with "partial support" (and explain what that is).